pkg/git/utils.go

```diff
@@ -0,0 +1,48 @@
+package git
+
+import (
+	"strings"
+)
+
+// CheckForValidForkRemote checks if the remote URL is a valid remote fork for the upstream URL.
+func CheckForValidForkRemote(upstreamURL, remoteURL, repo string) bool {
+	urlPart1, urlPart2 := extractCommonParts(upstreamURL, remoteURL)
+	if urlPart1 == "https://github.com" && urlPart2 == repo {
+		return true
+	}
+
+	return false
+}
+
+// extractCommonParts takes two Git URLs and returns the common prefix and suffix.
+func extractCommonParts(url1, url2 string) (string, string) {
+	// Split the URLs into segments
```

Comment:
Is there ever a chance that a string could be passed in that is empty or doesn't contain a `/`?

Reply:
@briandowns 
Actually, no, in the main.go file; we have this:

```
func autoForwardPort(c *cli.Context) {

	if ForkURL == "" {
		logrus.Fatalf("FORK environment variable must be set to run auto-forward-port")
	}
	
```
But I am adding an extra check before extractCommonParts.

Reply:
Not necessary if there's no chance.